packages/eslint-plugin/tests/rules/prefer-readonly-parameter-types.test.ts

```diff
@@ -222,6 +222,21 @@ ruleTester.run('prefer-readonly-parameter-types', rule, {
         },
       ],
     },
+    // PrivateIdentifier is handled without throwing error.
+    {
+      code: `
+        class Foo {
+          readonly #privateField = 'foo';
+          #privateMember() {}
+        }
+        function foo(arg: Foo) {}
+      `,
+      options: [
+        {
+          treatMethodsAsReadonly: true,
+        },
+      ],
+    },
```

Comment:
to double check - this case is ignored when `#privateField` is not `readonly` as well?
Could you add a test to verify this please? EG this should not error:

```ts
    {
      code: `
        class Foo {
          #privateField = 'foo';
          #privateMember() {}
        }
        function foo(arg: Foo) {}
      `,
      options: [
        {
          treatMethodsAsReadonly: false,
        },
      ],
    },
```

Reply:
Interesting.
For `private` fields - we 100% should be treating them as "public" (and thus subject them to `readonly` checks), however for `#private` fields, they should be exempt from `readonly` checks.

The intent behind the rule is that you should not be able to mutate the type at runtime. You can access `private` fields via computed member access syntax, but because you cannot access `#private` fields at all outside the class - they shouldn't be included in the checks.
([playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8CwAoaXoAcBOAlgG5gAuAproSRQAp7w7QC80AjANwbbQDEpeAK7l6xMpVYAmLugC+GDMHgA7CKWgFVpMMuAToy8gHc4iABQBKGZrU69AOnxi6DHB2gB6D9AAmBCGAgIPBG5D7QAEYAntAAKgDKADSRgupG8HgA1jBk0ELKpAQAtuQYNtq65ADaAORONCKuNQC67l7QgcGh4dFx8R3K4elZOer5hSUK6O1+AUEhYZExCdAAggByACK+8OQwyvBpGZkdY4IFxaXo5Xbk9vxCjc7k1lq3tQ-Cog0tHEA))

Reply:
Thanks for the playground. I didn't know about that. I just enjoy all your comments on any issues or prs. I think very constructive(very educational to me) :D

please check this [commit](https://github.com/typescript-eslint/typescript-eslint/pull/4343/commits/9a8d4f93d68f39db6af1e2c2372da11b9e273e59)